src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/LeonardoService.scala

```diff
@@ -1,14 +1,26 @@
 package org.broadinstitute.dsde.workbench.leonardo.service
 
+import akka.http.scaladsl.model.StatusCodes
 import com.typesafe.scalalogging.LazyLogging
 import org.broadinstitute.dsde.workbench.leonardo.dao.DataprocDAO
-import org.broadinstitute.dsde.workbench.leonardo.db.DbReference
-import org.broadinstitute.dsde.workbench.leonardo.model.{Cluster, ClusterRequest, ClusterResponse}
+import org.broadinstitute.dsde.workbench.leonardo.db.{DataAccess, DbReference}
+import org.broadinstitute.dsde.workbench.leonardo.model.ModelTypes.GoogleProject
+import org.broadinstitute.dsde.workbench.leonardo.model.{Cluster, ClusterRequest, ClusterResponse, LeoException}
+import slick.dbio.DBIO
 
 import scala.concurrent.{ExecutionContext, Future}
 
+case class ClusterNotFoundException(googleProject: GoogleProject, clusterName: String) extends LeoException(s"Cluster $googleProject/$clusterName not found")
```

Comment:
Consider this a proposal that we start using more descriptive exception classes. Insta-wrapping the exception into a `WorkbenchExceptionWithErrorReport` (so the akka-http route exception handler knows what to do with it) seems like an _okay_ mechanism but I'm not super fond of it.

Another option would be to add an optional status code to the `LeoException` base class (default 500), and let exception subclasses define their own (e.g. 404 for `ClusterNotFoundException`). We could then catch `LeoException` in the route exception handler, unpack the status code, and _possibly_ build an `ErrorReport` then instead.

Thoughts?

Reply:
I kind of prefer your second option: add a status code to `LeoException`, then catch `LeoException`s in the Leo  exception handler and build an `ErrorReport` there. Then `LeoException`s are clean in that they don't need to know about `ErrorReport`s -- the only place that has to deal with `ErrorReport` is the exception handler.

Just my 2c.

Reply:
I like the idea of descriptive exception classes, though.

Reply:
Great. We'll give this a shot and see how it works out. I have strong and sad feelings about Rawls' ErrorReport vs. exception classes and I've been jonesing for improvements there.